src/Microsoft.DotNet.XHarness.Common/CommandDiagnostics.cs

```diff
@@ -64,6 +75,12 @@ public class CommandDiagnostics : IDiagnosticsData
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
     public bool? IsDevice { get; set; }
 
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? WasmEngine { get; set; }
```

Comment:
I'd suggest to not add new fields but rather put the engine into the `Target`/`TargetOS` fields as you won't have to plumb these further to Kusto.

For instance, in emulators/phones, this would have the version of the Android and architecture.